Architecture.md

```diff
@@ -28,7 +28,29 @@
 
 #### Client applications deployment
 
-***TODO: [INFRA] fill for the architecture used to deploy clients' applications***
+Pomegranate, the application and deployment manager, is in charge of monitoring and managing the containers' lifecycle, as well as configuring them.
+
+The [Rust](https://www.rust-lang.org/) programming language is used to develop the service. We used it because
+we wanted an efficient and reliable language that could be trusted with this job thanks to its compiled nature and the restrictions it brings to drive us to write better code,
+and with support for the other technologies we use through the crates ecosystem.
```

Comment:
Define the crates ecosystem, explain what are crates and what Cargo is

Reply:
![image](https://github.com/paastech-cloud/docs/assets/24864674/00cb581e-159d-444f-b795-4762d234a7ed)
Is that enough ?

Reply:
Should be done